addons/web_editor/static/src/js/wysiwyg/wysiwyg.js

```diff
@@ -492,19 +492,52 @@ const Wysiwyg = Widget.extend({
      *
      * @param {object} [params]
      */
-    toggleLinkTools(params = {}) {
+    toggleLinkTools() {
         if (this.snippetsMenu) {
             if (this.linkTools) {
                 this.linkTools.destroy();
                 this.linkTools = undefined;
             } else {
                 const $btn = this.toolbar.$el.find('#create-link');
-                this.linkTools = new weWidgets.LinkTools(this, params, this.odooEditor.dom, $btn);
+                this.linkTools = new weWidgets.LinkTools(this, {}, this.odooEditor.dom, $btn);
                 this.linkTools.appendTo(this.toolbar.$el);
             }
         } else {
-            const linkDialog = new weWidgets.LinkDialog(this, params, this.$editable[0]);
+            const linkDialog = new weWidgets.LinkDialog(this, {}, this.$editable[0]);
             linkDialog.open();
+            linkDialog.on('save', this, (linkInfo) => {
+                const linkUrl = linkInfo.url;
+                const linkText = linkInfo.text;
+                const isNewWindow = linkInfo.isNewWindow;
+
+                const range = linkInfo.range;
+                const hasTextChanged = range.toString() !== linkText;
+
+                const ancestorAnchor = $(range.startContainer).closest('a')[0];
+                let anchors = [];
+                if (ancestorAnchor && ancestorAnchor === $(range.endContainer).closest('a')[0]) {
+                    anchors.push($(ancestorAnchor).html(linkText).get(0));
+                } else if (hasTextChanged) {
+                    const anchor = $('<A>' + linkText + '</A>')[0];
+                    range.insertNode(anchor);
+                    anchors.push(anchor);
+                } else {
+                    const anchor = $('a')[0];
+                    range.surroundContents(anchor);
+                    anchors.push(anchor);
+                }
+                for (const anchor of anchors) {
+                    $(anchor).attr('href', linkUrl);
+                    $(anchor).attr('class', linkInfo.className || null);
+                    $(anchor).css(linkInfo.style || {});
+                    if (isNewWindow) {
+                        $(anchor).attr('target', '_blank');
+                    } else {
+                        $(anchor).removeAttr('target');
+                    }
+                    range.selectNode(anchor);
+                };
+            });
```

Comment:
Looks ok. Have you tested well, made sure it works as expected?
I'm not a big fan of taking this out of link dialog because the dialog basically won't work without this piece of code so it better be absolutely necessary...

Reply:
It depends by what you mean by "work". It will still provide a link on "save". Which is the only thing that it was supposed to do on master. It's also what the class that extends it expect.

Reply:
It's only the wysiwyg who want to change the dom, other code can call the link dialog without having any document.

Reply:
Will it though? The link is created with this code right here. `save` will just return an object with information about a link, right?